### ISSUE TITLE
Handling WordPress and Git

### ISSUE THREAD

**Issue by dscotese** [voluntaryist/site#6]
We are working out how to handle WordPress here in the voluntaryist site.  Here are the elements that seem important to me:
- WP will keep itself updated.
- We need to track WP code changes
- There are subdirectories under the WordPress install directory where we do want to track changes.

**Comment by dooglio** [comment]
There is a SQL component to Wordpress, keep in mind. I assume that you will use mySql? If so, then you can just use the mysqlbackup command, dump out the SQL commands into a text file and keep a backup in git.

**Comment by dscotese** [comment]
That's right.  I'll make an issue for the creation of a Db Backup-to-git cron job.

**Comment by scbrown** [comment]
we might want to use git subtrees:
http://blogs.atlassian.com/2013/05/alternatives-to-git-submodule-git-subtree/

that way we can track wp's github while keeping our changes under that directory in our repo. 

``` shell
git remote add -f wp https://github.com/WordPress/WordPress.git
git subtree add --prefix wp wp master --squash
```

make any changes you need under the wp directory and commit them like you would normally. This is similar to the submodules approach I was talking about earlier. I don't have enough experience with either to know which would be the best to use for this case.

**Comment by dscotese** [comment]
I read through https://git-scm.com/book/en/v1/Git-Tools-Subtree-Merging.  I think that we could do this, and it would enable us to modify WP but still get their changes as they make them (the webserver does this automatically anyway with each new release, but doesn't track changes).  If someone notices that we made an excellent change to WP, they can go through WP's process to contribute to that project (but WP's Github project is read-only, I think).

We would have a branch specific to the wordpress folder for the voluntaryist.com-specific changes (like changes to files in the theme folder).

If that is how you (Steve) understand it too, let me know and I'll proceed.  Of course, anyone else can chime in too.  I'm on a learning curve here.

**Comment by dooglio** [comment]
This is a good thing, because you might want to tweak the theme, and
that would be code-changes that could be lost in an upgrade if you are
not careful.

**Comment by scbrown** [comment]
That's how I understand it. I tried a basic test yesterday and it seems to do what we need it to. We'll just have to see how the workflow will go.

**Comment by dscotese** [comment]
I made the changes and pushed them to master.  I also added a page to the Wiki for any one who wants to develop against a local copy of the site.  I want to leave this issue open until we have some modifications under the wp folder (like issue #3) that prove it's doing what we want.
